### PR TITLE
add upload view

### PR DESCRIPTION
**Pull request by jacksonargo**
* adds view for  `/sheets/upload` to users to upload sheet music.

### CODE REVIEW

File: api_server_test.go [hunk 1]
```diff
@@ -109,127 +112,359 @@ func Test_acceptsType(t *testing.T) {
...
+            <td colspan="2"><input type="file" name="upload_file" required="required" accept="application/pdf"/></td>
+        </tr>
+        <tr>
+            <td colspan="2"><input type="submit" value="submit"/></td>
+        </tr>
+        </tbody>
+    </table>
+</form>
+</body>
+</html>`
```
Comment (0az) [inline api_server_test.go]: Probably shouldn't be checking against a specific string.

Comment (jacksonargo) [reply api_server_test.go]: Why is that? This is what we should get, given the test inputs.

Comment (0az) [reply api_server_test.go]: A parsing solution would probably be more robust.

Comment (jacksonargo) [reply api_server_test.go]: Oh I see what you mean! Good call!

File: api_server.go [hunk 2]
```diff
@@ -131,45 +127,90 @@ func acceptsType(r *http.Request, mimeType string) bool {
...
+			ContentType: "application/pdf",
+			Name:        sheet.ObjectKey(),
+			Tags:        sheet.Tags(),
+			Buffer:      pdfBytes,
+		}
+		if err := x.PutObject(SheetsBucketName, &object); err != nil {
+			logger.WithError(err).Error("storage.PutObject() failed")
+			return nil, http.StatusInternalServerError
+		}
+		return nil, http.StatusOK
```
Comment (jacksonargo) [inline api_server.go]: This should redirect back to `/sheets` for web brrowsers

File: api_server.go [hunk 3]
```diff
@@ -64,23 +70,23 @@ func logRequest(handlerFunc APIHandlerFunc, r *http.Request) ([]byte, int) {
...
 		"request_seconds": time.Since(start).Seconds(),
-		"status_code":     code,
+		"status_code":     results.code,
 	}
 	switch true {
-	case code >= 500:
+	case results.code >= 500:
 		logger.WithFields(fields).Error("request failed")
-	case 400 <= code && code < 500:
+	case 400 <= results.code && results.code < 500:
```
Comment (0az) [inline api_server.go]:
```suggestion
	case results.code >= 400:
```
Slightly more readable.

**Pull request merged**